server/src/weaverbird/pipeline/pipeline.py

```diff
@@ -256,6 +256,63 @@ def remove_void_conditions_from_filter_steps(
     return final_steps
 
 
+def _is_empty(data: dict[str, Any] | list[Any]) -> bool:
+    if isinstance(data, dict):
+        return len(data) == 0
+    elif isinstance(data, list):
+        return len(data) == 0
+    return False
+
+
+def _remove_empty_elements_that_has_empty_value(
+    data: dict[str, Any] | list[Any]
+) -> dict[str, Any] | list[Any]:
+    """
+    This should delete all empty arrays and empty dict
+    """
+    if isinstance(data, dict):
+        return {
+            k: _remove_empty_elements_that_has_empty_value(v)
+            for k, v in data.items()
+            if not _is_empty(v)
+        }
+    elif isinstance(data, list):
+        return [
+            _remove_empty_elements_that_has_empty_value(item)
+            for item in data
+            if not _is_empty(item)
+        ]
```

Comment:
We might still have empty elements here: if `_remove_empty_elements_that_has_emtpy_value` returns an emtpy dict. Is that the desired behaviour ?